### PR TITLE
Bearer NTLM transformation support

### PR DESCRIPTION
**Pull request by kunagpal**

### CODE REVIEW

File: examples/v1.0.0/helpers.json [hunk 1]
```diff
@@ -68,6 +68,28 @@
 			"responses": [],
 			"synced": false
 		},
+    {
+      "id": "951fc3e8-c6b6-5c19-9f69-4e7499b3127f",
+      "headers": "Authorization: Bearer wkjehbxoqnunc2k3",
+      "url": "http://echo.getpostman.com/auth/bearer",
```
Comment (shamasis) [inline examples/v1.0.0/helpers.json]: Seems that the header has the auth repeated

Comment (kunagpal) [reply examples/v1.0.0/helpers.json]: Addressed in 67b33e7749dc7a0d4d2d0d55e2094fb5130da9b6.

File: examples/v2.0.0/helpers.json [hunk 2]
```diff
@@ -103,6 +103,42 @@
...
+            "request": {
+                "auth": {
+                    "type": "bearer",
+                    "bearer": {
+                      "token": "wkjehbxoqnunc2k3"
+                    }
+                },
+                "url": "http://echo.getpostman.com/auth/bearer",
+                "method": "GET",
+                "header": [
```
Comment (kamalaknn) [inline examples/v2.0.0/helpers.json]: Headers need to be removed here too.

File: examples/v2.1.0/helpers.json [hunk 3]
```diff
@@ -112,6 +112,44 @@
...
+                        "key": "token",
+                        "value": "wkjehbxoqnunc2k3",
+                        "type": "string"
+                    }]
+                },
+                "url": "http://echo.getpostman.com/auth/bearer",
+                "method": "GET",
+                "header": [
+                    {
+                        "key": "Authorization",
```
Comment (kamalaknn) [inline examples/v2.1.0/helpers.json]: Here too.

File: test/unit/fixtures/sample-auth.js [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+                id: '951fc3e8-c6b6-5c19-9f69-4e7499b3127g',
+                // eslint-disable-next-line max-len
+                headers: 'Authorization: NTLM TlRMTVNTUAADAAAAGAAYAFYAAAAYABgAbgAAAAAAAABIAAAADgAOAEgAAAAAAAAAVgAAAAAAAACGAAAABYKIogUBKAoAAAAPcABvAHMAdABtAGEAbgCDJfCgEQC+3wAAAAAAAAAAAAAAAAAAAAB6O2T1blvWpb/pqHtSdqcZ/A34nPBZe20=',
+                url: 'http://echo.getpostman.com/auth/ntlm',
+                preRequestScript: '',
+                method: 'GET',
+                data: [],
+                queryParams: [],
+                rawModeData: '',
+                headerData: [{
```
Comment (kamalaknn) [inline test/unit/fixtures/sample-auth.js]: It is okay to have it in the tests, but ideally, collection requests do not have auth and the header generated by the auth.

Comment (kunagpal) [reply test/unit/fixtures/sample-auth.js]: This ensures that header transformations are not affected by auth (as was also the case with the examples themselves).

**Pull request merged**